### ISSUE TITLE
[2.1.1-53-beta] Syncing very slow for anm009

### ISSUE THREAD

**Issue by AnnaSchrimpf** [SanaMobile/vaccine_registry#99]
The syncing for anm009, which is has 25 villages attached, is very slow, taking 5+ min on each login (not just initial) with very good wifi connection. Could you explain what the syncing time is a function of? If extra villages add a lot of extra time required we might reconsider the village selection we display.

**Comment by kepaik** [comment]
Yes, that is a lot of villages and would incur a complete check across all villages.

The login sync is particularly long because this is where it checks all records for updated fields, not just newly created records. The sync would need to download all the child information and all the encounters for all the children of that village. Having 25 villages is a lot of encounters.

Is 25 villages a normal number? 
I was under the impression it would only be 2-3 villages.

If it is going to be normal to have the entire sum of villages, then we would need to think about an alternative sync strategy.

**Comment by AnnaSchrimpf** [comment]
I'm slightly confused because right now look-up works across all records, across all ANMs and villages, correct? So I'm not sure I follow why the addition of villages to the list forces a higher syncing load? 

We have discussed this issue before, but we are expecting lists to be around 10-15 villages.

**Comment by kepaik** [comment]
I was not privy to the earlier discussions, but @ewinkler told me that it was decided that the search would only be across the children in the associated village and that was an agreeable tradeoff.

**Comment by AnnaSchrimpf** [comment]
I can currently lookup across villages and tablets, though. I created SJA000097 as anm004 in village 2. I switch tablet, login as anm007 select village 6, and can successfully lookup - hence my confusion.

**Comment by kepaik** [comment]
If it's on the same device, the data is stored locally and queryable. However, if you were to switch devices and log in as a anm007, the initial sync would not include children from village 2.

**Comment by AnnaSchrimpf** [comment]
I see - so all encounters that are connected to any login that ever logged into the tablet are searchable, even if a particular village was never selected within that login (if that makes sense). So if all of these are stored locally, do we need to be concerned about tablet storage issues at all?

**Comment by kepaik** [comment]
Yes, that is correct.
Storage size shouldn't be too much of a concern, unless things scale significantly and for several years.

**Comment by ewinkler** [comment]
I would echo what @kepaik said regarding storage size. It would take something on the order of several years worth of data before there may even be an issue.
